lib/elixir/lib/version.ex

```diff
@@ -370,27 +426,20 @@ defmodule Version do
   @doc since: "1.8.0"
   @spec parse_requirement!(String.t()) :: Requirement.t()
   def parse_requirement!(string) when is_binary(string) do
-    case Version.Parser.parse_requirement(string) do
-      {:ok, spec} ->
-        Requirement.new(string, spec)
-
-      :error ->
-        raise InvalidRequirementError, string
+    case parse_requirement(string) do
+      {:ok, requirement} -> requirement
+      :error -> raise InvalidRequirementError, string
     end
   end
 
   @doc """
-  Compiles a requirement to its internal representation with
-  `:ets.match_spec_compile/1` for faster matching.
+  Compiles a requirement to an internal representation that may optimize matching.
 
-  The internal representation is opaque and cannot be converted to external
-  term format and then back again without losing its properties (meaning it
-  can not be sent to a process on another node and still remain a valid
-  compiled match_spec, nor can it be stored on disk).
+  The internal representation is opaque.
   """
   @spec compile_requirement(Requirement.t()) :: Requirement.t()
   def compile_requirement(requirement) do
-    Requirement.compile(requirement)
+    requirement
   end
```

Comment:
I noticed that this function became a no-op after this PR.
It no longer compiles as its docs describes.
My use case is that I passed a string representation of a requirement, and thinking it will compile it to a VersionRequirement, but it does not. We certainly need a guard, but... I am wondering if we should deprecate this function.

Reply:
It should not be deprecated according to https://github.com/elixir-lang/elixir/pull/9807#discussion_r377611433.